packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```
@@ -114,7 +114,7 @@ const SliceBuilder: React.FC<SliceBuilderProps> = ({
   };
 
   const onTakingSliceCustomScreenshot = (file: Blob) => {
-    generateSliceCustomScreenshot(variation.id, component, setData, file);
+    generateSliceCustomScreenshot(variation.id, component, file);
```

how are we giving feedback to the user?

---

the saga is wrapped with a `withLoading` that can be queried in the component. The success is easy to check but you are right that the failure is not handled. I'll add a toaster message on error!

---

oh actually the error toast is there already, so you get feedback when it fails (and when it succeeds, you just see the screenshot being updated)